src/nlp/api.jl

```diff
@@ -736,37 +734,37 @@ function jac_op!(
 end
 
 """
-    J = jac_lin_op(nlp, x)
+    J = jac_lin_op(nlp)
 
-Return the linear Jacobian at `x` as a linear operator.
+Return the linear Jacobian as a linear operator.
 The resulting object may be used as if it were a matrix, e.g., `J * v` or
 `J' * v`.
 """
-function jac_lin_op(nlp::AbstractNLPModel{T, S}, x::AbstractVector) where {T, S}
-  @lencheck nlp.meta.nvar x
+function jac_lin_op(nlp::AbstractNLPModel{T, S}) where {T, S}
   Jv = S(undef, nlp.meta.nlin)
   Jtv = S(undef, nlp.meta.nvar)
-  return jac_lin_op!(nlp, x, Jv, Jtv)
+  return jac_lin_op!(nlp, Jv, Jtv)
 end
```

Comment:
Keep the old jac_lin_op, add a line in the docstring to say it's deprecated. Use the macro @deprecate.